tests/conftest.py

```diff
@@ -2598,3 +2598,160 @@ def ceph_toolbox(request):
+            # Create random namespace bucket name
+            rand_ns_bucket = create_unique_resource_name('aws', constants.MCG_NS_BUCKET)
+
+            # Create actual namespace bucket
+            mcg_obj.create_new_namespace_bucket(rand_ns_bucket, write_resource[i], read_resources[i])
+            created_ns_buckets.append(rand_ns_bucket)
+            check_ns_bucket_validity(mcg_obj, rand_ns_bucket, write_resource[i], read_resources[i])
+        return rand_ns_bucket
+
+    def ns_buckets_cleanup():
+        for ns_bucket in created_ns_buckets:
+            delete_ns_bucket(ns_bucket, mcg_obj)
+
+    request.addfinalizer(ns_buckets_cleanup)
+
+    return _create_ns_buckets
+
+
+def check_ns_resource_validity(mcg_obj, ns_resource_name, target_bucket_name, endpoint):
+    """
+    Check namespace resource validity
+    """
+    # Retrieve the NooBaa system information
+    system_state = mcg_obj.send_rpc_query('system_api', 'read_system').json().get('reply')
+
+    # Retrieve the correct namespace resource info
+    match_resource = [
+        ns_resource for ns_resource in system_state
+        .get('namespace_resources') if ns_resource.get('name') == ns_resource_name
+    ]
+    assert match_resource, f"The NS resource named {ns_resource_name} was not found"
+    actual_target_bucket = match_resource[0].get('target_bucket')
+    actual_endpoint = match_resource[0].get('endpoint')
+
+    assert actual_target_bucket == target_bucket_name, (
+        f"The NS resource named {ns_resource_name} got "
+        f"wrong target bucket {actual_target_bucket} ≠ {target_bucket_name}"
+    )
+    assert actual_endpoint == endpoint, (
+        f"The NS resource named {ns_resource_name} got wrong endpoint "
+        f"{actual_endpoint} ≠ {endpoint}"
+    )
+
+
+def check_ns_bucket_validity(mcg_obj, ns_bucket_name, write_resource, read_resources):
+    """
+    Check namespace bucket validity
+    """
+    # Retrieve the NooBaa system information
+    system_state = mcg_obj.send_rpc_query('system_api', 'read_system').json().get('reply')
+
+    # Retrieve the correct namespace bucket info
+    match_buckets = [
+        ns_bucket for ns_bucket in system_state
+        .get('buckets') if ns_bucket.get('name') == ns_bucket_name
+    ]
+    assert match_buckets, f"The NS bucket named {ns_bucket_name} was not found"
+    ns_properties = match_buckets[0].get('namespace')
+    actual_read_resources = ns_properties.get('read_resources')
+    actual_write_resource = ns_properties.get('write_resource')
+
+    assert actual_read_resources == read_resources, (
+        f"The NS bucket named {ns_bucket_name} "
+        f"got wrong read resources {actual_read_resources} ≠ {read_resources}"
+    )
+    assert actual_write_resource == write_resource, (
+        f"The NS bucket named {ns_bucket_name} "
+        f"got wrong write resource {actual_write_resource} ≠ {write_resource}"
+    )
+
+
+def delete_ns_connection(ns_connection_name, mcg_obj):
```

Comment:
This should not be in conftest. I suggest to move it to ocs_ci/ocs/resources/objectbucket.py or create a new mcg-namespace resource file.

Reply:
moved to mcg.py as well